keepalived/vrrp/vrrp_print.c

```diff
@@ -418,3 +421,182 @@ vrrp_print_stats(void)
+		json_object_object_add(json_data, "fd_in",
+			json_object_new_int(vrrp->fd_in));
+		json_object_object_add(json_data, "fd_out",
+			json_object_new_int(vrrp->fd_out));
+		json_object_object_add(json_data, "debug",
+			json_object_new_int(vrrp->debug));
+		json_object_object_add(json_data, "quick_sync",
+			json_object_new_int(vrrp->quick_sync));
+		json_object_object_add(json_data, "version",
+			json_object_new_int(vrrp->version));
+		json_object_object_add(json_data, "smtp_alert",
+			json_object_new_boolean(vrrp->smtp_alert));
+		json_object_object_add(json_data, "notify_exec",
+			json_object_new_boolean(vrrp->notify_exec));
+		json_object_object_add(json_data, "ms_down_timer",
+			json_object_new_int(vrrp->ms_down_timer));
+		json_object_object_add(json_data, "sands",
+			json_object_new_int(vrrp->sands.tv_sec));
+		json_object_object_add(json_data, "auth_type",
+			json_object_new_int(vrrp->auth_type));
+		json_object_object_add(json_data, "ip_id",
+			json_object_new_int(vrrp->ip_id));
+
+		// Dump stats to json
+		json_object_object_add(json_stats, "advert_rcvd",
+			json_object_new_int64(vrrp->stats->advert_rcvd));
+		json_object_object_add(json_stats, "advert_sent",
+			json_object_new_int(vrrp->stats->advert_sent));
+		json_object_object_add(json_stats, "become_master",
+			json_object_new_int(vrrp->stats->become_master));
+		json_object_object_add(json_stats, "release_master",
+			json_object_new_int(vrrp->stats->release_master));
+		json_object_object_add(json_stats, "packet_len_err",
+			json_object_new_int64(vrrp->stats->packet_len_err));
+		json_object_object_add(json_stats, "advert_interval_err",
+			json_object_new_int64(vrrp->stats->advert_interval_err));
+		json_object_object_add(json_stats, "ip_ttl_err",
+			json_object_new_int64(vrrp->stats->ip_ttl_err));
+		json_object_object_add(json_stats, "invalid_type_rcvd",
+			json_object_new_int64(vrrp->stats->invalid_type_rcvd));
+		json_object_object_add(json_stats, "addr_list_err",
+			json_object_new_int64(vrrp->stats->addr_list_err));
+		json_object_object_add(json_stats, "invalid_authtype",
+			json_object_new_int(vrrp->stats->invalid_authtype));
+		json_object_object_add(json_stats, "authtype_mismatch",
+			json_object_new_int(vrrp->stats->authtype_mismatch));
+		json_object_object_add(json_stats, "auth_failure",
+			json_object_new_int(vrrp->stats->auth_failure));
+		json_object_object_add(json_stats, "pri_zero_rcvd",
+			json_object_new_int64(vrrp->stats->pri_zero_rcvd));
+		json_object_object_add(json_stats, "pri_zero_sent",
+			json_object_new_int64(vrrp->stats->pri_zero_sent));
+
+		// Add both json_data and json_stats to main instance_json
+		json_object_object_add(instance_json, "data", json_data);
+		json_object_object_add(instance_json, "stats", json_stats);
+
+		// Add instance_json to main array
+		json_object_array_add(array, instance_json);
+
+	}
+	fprintf(file, "%s", json_object_to_json_string(array));
+	fclose(file);
+}
+#endif
```

Comment:
The following items are probably not worth reporting, since they only have meaning internally to keepalived:
       vrrp->debug
       vrrp->evip_add_ipv6
       vrrp->fd_in
       vrrp->fd_out
       vrrp->garp_pending
       vrrp->garp_refresh_timer
       vrrp->gna_pending
       vrrp->iname
       vrrp->init_state
       vrrp->ip_id
       vrrp->iptable_rules_set
       vrrp->ms_down_timer
       vrrp->notify_exec
       vrrp->preempt_time
       vrrp->quick_sync
       vrrp->sands
       vrrp->vipset
       vrrp->vmac_flags
       vrrp->vmac_ifname
       vrrp->wantstate

The following items are reported in the standard print output, and are probably worth including in the JSON output
       vrrp->adver_int
       vrrp->auth_data
       vrrp->evip
       vrrp->ifp->ifname
       if (vrrp->ifp->vmac) if_get_by_ifindex(vrrp->ifp->base_ifindex)->ifname
       vrrp->lower_prio_no_advert
       vrrp->master_saddr
       vrrp->saddr
       vrrp->script
       vrrp->script_backup
       vrrp->script_fault
       vrrp->script_master
       vrrp->script_stop
       vrrp->track_ifp
       vrrp->track_script
       vrrp->vroutes
       vrrp->vrules